src/main/java/frc/robot/commands/TestVision.java

```diff
@@ -0,0 +1,104 @@
+package frc.robot.commands;
+
+import java.util.Optional;
+
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Robot;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.util.Vision;
+
+public class TestVision extends CommandBase{
+  private Drivetrain m_drive;
+  private double encoderStart;
+  private Pose2d startPose;
+  private double encoderPosition;
+  private Pose2d currentPose = null;
+  private int endCounter = 0;
+  private int printCounter=0;
+  private double m_speed;
+
+  //How many frames it has to not see anything to end the command
+  private final int endDelay = 5;
+
+  //How many frames it will wait between prints
+  private final int printDelay = 50;
+
+  /**
+   * A command that moves and prints out distances
+   * @param speed What speed to move at
+   * @param drive The drivetrain
+   */
+  public TestVision(double speed, Drivetrain drive){
+    addRequirements(drive);
+    m_drive=drive;
+    m_speed=speed;
+  }
+
+  private double getDist(){
+    // return 0;
+    return m_drive.getLeftDistance()/2+m_drive.getRightDistance()/2;
+  }
+
+  private Pose2d getPose(){
+    Optional<Pair<Pose3d, Double>> p = Vision.getEstimatedGlobalPose(currentPose==null?m_drive.getPose():currentPose);
+    if(p.isPresent() && p.get().getFirst() != null && p.get().getSecond() != null && p.get().getFirst().getX() > -10000 && p.get().getSecond() >= 0){
```

Comment:
This code seems pretty similar to code before, checking for null stuff can be handled in vision.java and just be part of getEstimatedGlobalPose

Reply:
Maybe, but I might want other information, or I might not want it to return null somewhere else.

Reply:
I created a new function in Vision.java. I can't change the other one because the drivetrain needs the latency.